pylops/signalprocessing/dct.py

```diff
@@ -0,0 +1,96 @@
+__all__ = ["DCT"]
+
+from typing import List, Optional, Union
+
+import numpy as np
+from scipy import fft
+
+from pylops import LinearOperator
+from pylops.utils._internal import _value_or_sized_to_tuple
+from pylops.utils.decorators import reshaped
+from pylops.utils.typing import DTypeLike, InputDimsLike, NDArray
+
+
+class DCT(LinearOperator):
+    r"""Discrete Cosine Transform.
+
+    Apply 1D or ND-Cosine Transform along one or more ``axes`` of a multi-dimensional
+    array of size ``dims``.
+
+    This operator is an overload of :func:`scipy.fft.dctn` in forward mode and :func:`scipy.fft.idctn`
+    in adjoint mode.
+
+    Parameters
+    ----------
+    dims : :obj:`int` or :obj:`tuple`
+        Number of samples for each dimension
+    type : :obj:`int`, optional
+        Type of DCT (see scipy's documentation for more details). Default type is 2.
```

Comment:
Here it would be nice to link directly to the docs. For example, see here: https://github.com/PyLops/pylops/blob/master/pylops/signalprocessing/fft2d.py#L241

Reply:
updated in [0ee0b44](https://github.com/PyLops/pylops/pull/480/commits/0ee0b44364f26ca3f192b968d793f41582dca77c)